src/ImageRunModal.jsx

```diff
@@ -633,6 +588,77 @@ export class ImageRunModal extends React.Component {
         return owner === systemOwner;
     };
 
+    isFormValid = validationFailed => Object.keys(validationFailed).length == 0;
+
+    async validateContainerName(containerName) {
+        try {
+            await client.containerExists(this.isSystem(), containerName);
+        } catch (error) {
+            return;
+        }
+        return _("Name already in use");
+    }
+
+    async validateForm() {
+        const { publish, volumes, env, containerName } = this.state;
+        const validationFailed = { };
+
+        const publishValidation = publish.map(a => {
+            return {
+                IP: validatePublishPort(a.IP, "IP"),
+                hostPort: validatePublishPort(a.hostPort, "hostPort"),
+                containerPort: validatePublishPort(a.containerPort, "containerPort"),
+            };
+        });
+        if (publishValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.publish = publishValidation;
+
+        const volumesValidation = volumes.map(a => {
+            return {
+                hostPath: validateVolume(a.hostPath, "hostPath"),
+                containerPath: validateVolume(a.containerPath, "containerPath"),
+            };
+        });
+        if (volumesValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.volumes = volumesValidation;
+
+        const envValidation = env.map(a => {
+            return {
+                envKey: validateEnvVar(a.envKey, "envKey"),
+                envValue: validateEnvVar(a.envValue, "envValue"),
+            };
+        });
+        if (envValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.env = envValidation;
+
+        const containerNameValidation = await this.validateContainerName(containerName);
+
+        if (containerNameValidation)
+            validationFailed.containerName = containerNameValidation;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231018-100152-f200e65a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)